torchao/dtypes/nf4tensor.py

```diff
@@ -188,6 +221,7 @@ def __init__(
         self.scaler_mean = scaler_mean
         self.quantized_data = quantized_data
         self.nf4 = nf4
+        self.transpose = transpose
```

Comment:
If we are doing the lazy transpose you need to update the unlatten and flatten methods
https://github.com/drisspg/transformer_nuggets/pull/24/files#diff-6954806b06a395823616c5a0fefc77b5cf274b3f88a5381c38856cbf74d6f2bfR420

Reply:
Hm, maybe I'll put this into the strides instead and rely on `is_contiguous` instead.